DB/src/main/java/io/deephaven/db/v2/parquet/ParquetTableWriter.java

```diff
@@ -259,14 +210,24 @@ public static void write(Table t, TableDefinition definition, String path, Map<S
     }
 
     private static Table pretransformTable(Table t, TableDefinition definition) {
-        List<String> columnsTransform = new ArrayList<>();
-        for (ColumnDefinition column : definition.getColumns()) {
-            if (StringSet.class.isAssignableFrom(column.getDataType())) {
-                columnsTransform.add(column.getName() + " = " + column.getName() + ".values()");
+        List<String> updateViewColumnsTransform = new ArrayList<>();
+        List<String> viewColumnsTransform = new ArrayList<>();
+        for (ColumnDefinition<?> column : definition.getColumns()) {
+            final String colName = column.getName();
+            if (t.hasColumns(colName)) {
+                if (StringSet.class.isAssignableFrom(column.getDataType())) {
+                    updateViewColumnsTransform.add(colName + " = " + colName + ".values()");
+                }
+                viewColumnsTransform.add(colName);
+            } else {
+                viewColumnsTransform.add(colName + " = " + TableTools.nullTypeAsString(column.getDataType()));
             }
         }
-        if (columnsTransform.size() > 0) {
-            t = t.updateView(columnsTransform.toArray(new String[0]));
+        if (viewColumnsTransform.size() > 0) {
+            t = t.view(viewColumnsTransform.toArray((new String[0])));
```

Comment:
We have constants for empty string arrays and empty select column arrays:
* `io.deephaven.datastructures.util.CollectionUtil#ZERO_LENGTH_STRING_ARRAY`
* `io.deephaven.db.v2.select.SelectColumn#ZERO_LENGTH_SELECT_COLUMN_ARRAY`

Reply:
Yeah and it doesn't matter, the optimizer is going to make that code disappear anyway, we are making the code harder to read and pulling more imports for no good reason.

Reply:
It shuts IntelliJ up.